src/modules/27-platform/secrets/components/SecretReference/__test__/SecretReference.test.tsx

```diff
@@ -6,70 +6,291 @@
+    // Check if secret from response is appearing on screen
+    const secretName = screen.getByText('text1')
+    expect(secretName).toBeInTheDocument()
   })
 
-  test('render with secret type file', async () => {
+  test('render with type as SecretFile', async () => {
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: false,
+        error: null,
+        data: mockData.data,
+        refetch: jest.fn().mockReturnValue(mockData.data)
+      })
+    )
+    ;(listSecretsV2Promise as jest.Mock).mockImplementation(listSecretsV2PromiseMock)
+
     const { container } = render(
       <TestWrapper>
-        <SecretReference type="SecretFile" accountIdentifier="dummy" mock={mockData as any} onSelect={noop} />
+        <SecretReference type="SecretFile" accountIdentifier="dummy" orgIdentifier="testOrg" onSelect={noop} />
       </TestWrapper>
     )
     await waitFor(() => getByText(container, 'entityReference.apply'))
-    expect(getByText(container, 'account')).toBeTruthy()
-    expect(container).toMatchSnapshot()
+
+    // Check if correct tabs are getting rendered
+    const projectTab = screen.queryByText('projectLabel')
+    const orgTab = screen.getByText('orgLabel')
+    const accountTab = screen.getByText('account')
+    expect(projectTab).not.toBeInTheDocument()
+    expect(orgTab).toBeInTheDocument()
+    expect(accountTab).toBeInTheDocument()
+    // Check if secret from response is appearing on screen
+    const secretName = screen.getByText('text1')
+    expect(secretName).toBeInTheDocument()
   })
 
   test('render with no secret type', async () => {
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: false,
+        error: null,
+        data: mockData.data,
+        refetch: jest.fn().mockReturnValue(mockData.data)
+      })
+    )
+    ;(listSecretsV2Promise as jest.Mock).mockImplementation(listSecretsV2PromiseMock)
+
     const { container } = render(
       <TestWrapper>
-        <SecretReference accountIdentifier="dummy" mock={mockData as any} onSelect={noop} />
+        <SecretReference
+          accountIdentifier="testAccount"
+          orgIdentifier="testOrg"
+          projectIdentifier="testProject"
+          onSelect={noop}
+        />
       </TestWrapper>
     )
     await waitFor(() => getByText(container, 'entityReference.apply'))
-    expect(getByText(container, 'account')).toBeTruthy()
-    expect(container).toMatchSnapshot()
+
+    // Check if correct tabs are getting rendered
+    const projectTab = screen.getByText('projectLabel')
+    const orgTab = screen.getByText('orgLabel')
+    const accountTab = screen.getByText('account')
+    expect(projectTab).toBeInTheDocument()
+    expect(orgTab).toBeInTheDocument()
+    expect(accountTab).toBeInTheDocument()
+    // Check if secret from response is appearing on screen
+    const secretName = screen.getByText('text1')
+    expect(secretName).toBeInTheDocument()
```

Comment:
Might as well inline these assertions instead of extracting them into variables first.
```suggestion
    expect(screen.getByText('projectLabel')).toBeInTheDocument()
    expect(screen.getByText('orgLabel')).toBeInTheDocument()
    expect(screen.getByText('account')).toBeInTheDocument()

    // Check if secret from response is appearing on screen
    expect(screen.getByText('text1')).toBeInTheDocument()
```